community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/SpatialCRSSchemaIndex.java

```diff
@@ -106,11 +110,11 @@ public SpatialCRSSchemaIndex( IndexDescriptor descriptor,
         indexFile = new File( indexDir.directoryForIndex( indexId ), "index-" + indexId );
         if ( crs.getDimension() == 2 )
         {
-            curve = new HilbertSpaceFillingCurve2D( envelopeFromCRS( crs ), 8 );
+            curve = new HilbertSpaceFillingCurve2D( envelopeFromCRS( crs ), maxLevels );
```

Comment:
I don't see where there is a guard against using a new setting of maxLevels for an already existing index. Also no testcase for this either.

Reply:
Yes, this is planned for a future PR, the work for making extents configurable also. In the meantime the fact that the settings are all _unsupported_ is the only protection.